src/index_factory.cpp

```diff
@@ -88,19 +89,53 @@ std::unique_ptr<AbstractGraphStore> IndexFactory::construct_graphstore(const Gra
     }
 }
 
+template <typename T>
+std::shared_ptr<PQDataStore<T>> IndexFactory::construct_pq_datastore(DataStoreStrategy strategy, size_t num_points,
+                                                                     size_t dimension, Metric m, size_t num_pq_chunks,
+                                                                     bool use_opq)
+{
+    std::shared_ptr<Distance<T>> distance_fn;
+    std::shared_ptr<QuantizedDistance<T>> quantized_distance_fn;
+
+    quantized_distance_fn = std::make_shared<PQL2Distance<T>>((uint32_t)num_pq_chunks, use_opq);
+    switch (strategy)
+    {
+    case DataStoreStrategy::MEMORY:
+        distance_fn.reset(construct_inmem_distance_fn<T>(m));
+        return std::make_shared<diskann::PQDataStore<T>>(dimension, (location_t)(num_points), num_pq_chunks,
+                                                         distance_fn, quantized_distance_fn);
+    default:
+        // REFACTOR TODO: We do support diskPQ - so we may need to add a new class for SSDPQDataStore!
+        break;
+    }
+    return nullptr;
+}
+
 template <typename data_type, typename tag_type, typename label_type>
 std::unique_ptr<AbstractIndex> IndexFactory::create_instance()
 {
     size_t num_points = _config->max_points + _config->num_frozen_pts;
     size_t dim = _config->dimension;
+    // auto graph_store = construct_graphstore(_config->graph_strategy, num_points);
+    auto data_store = construct_datastore<data_type>(_config->data_strategy, num_points, dim, _config->metric);
+    std::shared_ptr<AbstractDataStore<data_type>> pq_data_store = nullptr;
+
+    if (_config->data_strategy == DataStoreStrategy::MEMORY && _config->pq_dist_build) {
+        pq_data_store = construct_pq_datastore<data_type>(_config->data_strategy, num_points + _config->num_frozen_pts, dim, _config->metric,
+                                                          _config->num_pq_chunks, _config->use_opq);
+    } else {
+        pq_data_store = data_store;
```

Comment:
if it is not a pq_dist_build, can we just pass this as  nullptr, why point it to data_store ?